gui/map.cc

```diff
@@ -69,28 +76,57 @@ Map::Map(QWidget *parent,
   manager_ = new QNetworkAccessManager(this);
   connect(this,SIGNAL(loadFinished(bool)),
 	  this,SLOT(loadFinishedX(bool)));
-  this->logTimeX("Start map constuctor");
+  this->logTime("Start map constuctor");
+
+#if HAVE_WEBENGINE
+  MarkerClicker *mclicker = new MarkerClicker(this);
+  QWebChannel* channel = new QWebChannel(this->page());
+  this->page()->setWebChannel(channel);
+  // Note: A current limitation is that objects must be registered before any client is initialized.
+  channel->registerObject(QStringLiteral("mclicker"), mclicker);
+  connect(mclicker, SIGNAL(markerClicked(int, int )), this, SLOT(markerClicked(int, int)));
+  connect(mclicker, SIGNAL(logTime(const QString &)), this, SLOT(logTime(const QString &)));
+#endif
+
   QString baseFile =  QApplication::applicationDirPath() + "/gmapbase.html";
   if (!QFile(baseFile).exists()) {
     QMessageBox::critical(0, appName,
 			  tr("Missing \"gmapbase.html\" file.  Check installation"));
   }
   else {
     QString urlStr = "file:///" + baseFile;
-    load(QUrl(urlStr));
+    this->load(QUrl(urlStr));
   }
+
+#ifdef DEBUG_JS_GENERATION
+  dbgdata_ = new QFile("mapdebug.js");
+  if (dbgdata_->open(QFile::WriteOnly | QIODevice::Truncate)) {
+    dbgout_ = new QTextStream(dbgdata_);
+  }
+#endif
+
 }
 
 //------------------------------------------------------------------------
 Map::~Map()
 {
   if (busyCursor_)
     QApplication::restoreOverrideCursor();
+#ifdef DEBUG_JS_GENERATION
+  if (dbgout_) {
+    delete dbgout_;
```

Comment:
OPtional: If we embed those in the Map object instead of making them raw pointers in the Map Object, they'll self-destruct. Even cleaner than smart pointers.